client/tokenized-payment-request/index.js

```diff
@@ -60,17 +60,11 @@ jQuery( ( $ ) => {
 		productData: getPaymentRequestData( 'product' ) || undefined,
 	} );
 
-	// We don't need to initialize payment request on the checkout page now because it will be initialized by updated_checkout event.
-	if (
-		getPaymentRequestData( 'button_context' ) !== 'checkout' ||
-		getPaymentRequestData( 'button_context' ) === 'pay_for_order'
-	) {
-		wooPaymentsPaymentRequest.init();
-	}
+	wooPaymentsPaymentRequest.init();
 
-	// We need to refresh payment request data when total is updated.
+	// When the cart is updated, the PRB is removed from the page and needs to be re-initialized.
 	$( document.body ).on( 'updated_cart_totals', () => {
-		doAction( 'wcpay.payment-request.update-button-data' );
+		wooPaymentsPaymentRequest.init();
```

Comment:
@frosso, in the PDP implementation, [the  `wcpay.payment-request.update-button-data` hook](https://github.com/Automattic/woocommerce-payments/blob/develop/client/tokenized-payment-request/payment-request.js#L147-L184) appears to be doing a bit more work than simply re-initialising the payment request. I _think_ this should be fine and appears to be working as expected for me, but did you previously use this hook as a placeholder or was there some greater purpose that might be lost by simply replacing it with the re-init? 🤔

Reply:
Thanks for flagging that - I don't think I noticed this change in my earlier review.

The hook is meant to check [if shipping requirements changed](https://github.com/Automattic/woocommerce-payments/blob/f8a38f7f68cebdc036aa720c5ae9d1d4dd2c04db/client/tokenized-payment-request/payment-request.js#L169-L169) when new products (or new options) have been added.

Doing `wooPaymentsPaymentRequest.init();`, instead, skips that step.

I could see this potentially being an issue for plugins like "Order Bump for WooCommerce", when a "simple" product is added to a cart only containing virtual products (e.g.: having the `woo-album` in the cart and adding `woo-beanie` later).
That said, I tested it locally and didn't see an issue with it. I simply copied the business logic [from the previous implementation](https://github.com/Automattic/woocommerce-payments/blob/f8a38f7f68cebdc036aa720c5ae9d1d4dd2c04db/client/payment-request/index.js#L471-L491).

@mdmoore what was the reason behind this change?

Reply:
Thank you for the explanation, that makes sense! I didn't realize that the `updated_cart_totals` would remove the PRB from the page 🤦